File: bindings/python/python/maxt/examples/bithumb_provider.py
"""Read public Bithumb market warnings, notices, and transfer fees."""

from __future__ import annotations

import asyncio

from maxt import BithumbAdapter


async def main() -> None:
    adapter = BithumbAdapter()
    warnings = await adapter.market_warnings()
    notices = await adapter.notices(5)
    fees = await adapter.transfer_fees("BTC")
    print(f"{len(warnings)} warning rows, {len(notices)} notices, {len(fees)} BTC fee rows")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/bithumb_provider.rs
//! Read Bithumb-specific public market metadata.
//!
//! ```text
//! cargo run --example bithumb_provider
//! ```

use maxt::adapters::BithumbAdapter;

#[tokio::main(flavor = "current_thread")]
async fn main() -> maxt::Result<()> {
    let adapter = BithumbAdapter::new();
    let warnings = adapter.market_warnings().await?;
    let notices = adapter.notices(Some(5)).await?;
    let fees = adapter.transfer_fees("BTC").await?;
    println!(
        "{} warning rows, {} notices, {} BTC fee rows",
        warnings.len(),
        notices.len(),
        fees.len()
    );
    Ok(())
}
